### PR TITLE
gen-license: init at 0.1.2

### PR DESCRIPTION
**Pull request by ryanccn**
###### Description of changes

https://github.com/nexxeln/license-generator

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -1671,6 +1671,8 @@ with pkgs;
 
   gen6dns = callPackage ../tools/networking/gen6dns { };
 
+  gen-license = callPackage ../development/tools/gen-license { };
+
```
Comment (getchoo) [inline pkgs/top-level/all-packages.nix:1675]:
```suggestion
  gen-license = callPackage ../development/tools/gen-license {
    inherit (darwin.apple_sdk.frameworks) Security;
  };
```

File: pkgs/development/tools/gen-license/default.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  cargoLock.lockFile = "${src}/Cargo.lock";
+
+  buildInputs =
+    [ ]
+    ++ lib.optionals stdenv.isDarwin [
+      darwin.apple_sdk_11_0.frameworks.CoreFoundation
+      darwin.apple_sdk_11_0.frameworks.Security
+      darwin.IOKit
+      libiconv
+    ];
```
Comment (getchoo) [inline pkgs/development/tools/gen-license/default.nix:28]:
```suggestion
  buildInputs = lib.optional stdenv.isDarwin [ Security ];
```
i was able to build without the other inputs on x86_64-darwin

File: pkgs/development/tools/gen-license/default.nix [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+  nativeBuildInputs = [ pkg-config ];
+
```
Comment (getchoo) [inline pkgs/development/tools/gen-license/default.nix:31]:
```suggestion
```
not needed for apple frameworks

File: pkgs/development/tools/gen-license/default.nix [hunk 4]
```diff
@@ -0,0 +1,38 @@
+{ lib
+, stdenv
+, rustPlatform
+, fetchCrate
+, darwin
+, libiconv
+, pkg-config
+}:
```
Comment (getchoo) [inline pkgs/development/tools/gen-license/default.nix:8]:
```suggestion
{ lib
, stdenv
, rustPlatform
, fetchCrate
, Security
}:
```

**Review by getchoo — CHANGES_REQUESTED** [review]

File: pkgs/development/tools/gen-license/default.nix [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  version = "0.1.2";
+
+  src = fetchCrate {
+    inherit pname version;
+    sha256 = "sha256-YZcycLQ436cjr2YTT7TEyMdeLTOl9oEfa5x3lgnnYyo=";
+  };
+
+  cargoLock.lockFile = "${src}/Cargo.lock";
+
+  buildInputs = lib.optional stdenv.isDarwin [ Security ];
```
Comment (getchoo) [inline pkgs/development/tools/gen-license/default.nix:19]:
```suggestion
  buildInputs = lib.optional stdenv.isDarwin Security;
```
oops! this is my bad (causes nested lists without this change)

**Comment by getchoo** [comment]
Result of `nixpkgs-review pr 244600` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>gen-license</li>
  </ul>
</details>

**Review by getchoo — APPROVED** [review]

**Review by emilylange — CHANGES_REQUESTED** [review]
Welcome to nixpkgs!

Looks good, just one thing:
Could you please split the change to `maintainer-list.nix` into its own separate commit?

Meaning two commits in total:
1. `maintainers: add ryanccn`
1. `gen-license: init at 0.1.2`

Thanks!

See https://nixos.org/manual/nixpkgs/unstable/#var-meta-maintainers for reference :)

**Pull request merged**